In [29]:
import numpy as np
import pandas as pd
from SPLIT_DATA import TrainTestSplit_FromSCRATCH
from sklearn.model_selection import train_test_split

In [30]:
data = pd.read_csv("HousePrice.csv")

In [31]:
data["Parking"].replace(["True" , "False"]  , [1 , 0] , inplace=True)
data["Warehouse"].replace(["True" , "False"]  , [1 , 0] , inplace=True)
data["Elevator"].replace(["True" , "False"]  , [1 , 0] , inplace=True)
data.drop("Address" , axis=1 , inplace=True)
X = np.array(data[["Area"]])
Y = np.array(data[["Price"]])
X.shape , Y.shape
# X.shape[0] == len(X) = 3473

((3473, 1), (3473, 1))

In [32]:
test_ratio = 0.4
shuffle  = True
split_data = TrainTestSplit_FromSCRATCH(test_ratio , shuffle )
x_train , x_test , y_train , y_test = split_data.split(X,Y)
print (x_train.shape )
print (x_test.shape )
print(y_train.shape)
print(y_test.shape)

(2084, 1)
(1389, 1)
(2084, 1)
(1389, 1)


In [33]:
x_train , x_test , y_train , y_test = train_test_split(X,Y , test_size=0.4)

print (x_train.shape )
print (x_test.shape )
print(y_train.shape)
print(y_test.shape)

(2083, 1)
(1390, 1)
(2083, 1)
(1390, 1)


|SPLITED PARTS|   X_TRAIN  | X_TEST | Y_TRAIN  | Y_TEST |
|------------------| --------- | --------- | ---------- | --------- |
MY_TRAIN_TEST_SPLIT ( test_size=0.4 )| (2084, 1)  | (1389, 1)  | (2084, 1)  | (1389, 1)  |
sklearn train_test_split  ( test_size=0.4 )| (2083, 1)  | (1390, 1)  | (2083, 1)  | (1390, 1)  |
